### ISSUE TITLE
Dialyzer Errors

### ISSUE THREAD

**Issue by maennchen** [elixir-cldr/cldr_dates_times#10]
```
lib/cldr/backend/date_time_formatter.ex:117:pattern_match
The pattern can never match the type.

Pattern:
{:error, 95}

Type:
{:error, {_, :datetime_format_lexer, {_, _}}, _}
________________________________________________________________________________
lib/cldr/backend/date_time_formatter.ex:183:call
The function call will not succeed.

Cldr.Substitution.substitute(binary(), [<<71, 77, 84>>, 0])

breaks the contract
([term(), ...], [String.t() | integer(), ...]) :: [term(), ...]
________________________________________________________________________________
lib/cldr/backend/date_time_formatter.ex:183:call
The function call will not succeed.

Cldr.Substitution.substitute(binary(), [<<85, 84, 67>>, 0])

breaks the contract
([term(), ...], [String.t() | integer(), ...]) :: [term(), ...]
________________________________________________________________________________
lib/cldr/backend/date_time_formatter.ex:183:call
The function call will not succeed.

Cldr.Substitution.substitute(binary(), [<<71, 77, 84>>, 0])

breaks the contract
([term(), ...], [String.t() | integer(), ...]) :: [term(), ...]
________________________________________________________________________________
lib/cldr/backend/date_time_formatter.ex:194:call
The function call will not succeed.

Cldr.Substitution.substitute(binary(), [<<71, 77, 84>>, 0])

breaks the contract
([term(), ...], [String.t() | integer(), ...]) :: [term(), ...]
________________________________________________________________________________
lib/cldr/backend/date_time_formatter.ex:194:call
The function call will not succeed.

Cldr.Substitution.substitute(binary(), [<<85, 84, 67>>, 0])

breaks the contract
([term(), ...], [String.t() | integer(), ...]) :: [term(), ...]
________________________________________________________________________________
lib/cldr/backend/date_time_formatter.ex:194:call
The function call will not succeed.

Cldr.Substitution.substitute(binary(), [<<71, 77, 84>>, 0])

breaks the contract
([term(), ...], [String.t() | integer(), ...]) :: [term(), ...]
```

Installed packages:
```
ex_cldr_dates_times  2.1.0
ex_cldr_numbers      2.6.4
ex_cldr_units        2.5.1
```

**Comment by kipcole9** [comment]
I apologise, that's sloppy of me.  I will work on it this week (I'm going to be out of internet reach for a few days but will work on it).

**Comment by maennchen** [comment]
@kipcole9 Thanks a lot. Would you be interested in a Travis / Whatever CI you like setup for the CLDR repos?

**Comment by kipcole9** [comment]
I actually have a Jenkins server I run my code through but I haven’t enabled dialyzer on it.  I've made a lot of progress in my understanding of dialyzer and I think all my other libs are clean now.

I will run dialyzer on my CI server from now as you are very politely suggesting!

Regards, —Kip

**Comment by kipcole9** [comment]
Sorry this is taking so long, am in a location with no internet until Monday so I'll keep working on this offline and aim to have it resolved early next week.

**Comment by maennchen** [comment]
@kipcole9 No worries... Thanks for your work!

**Comment by kipcole9** [comment]
Published [version 2.2.0](https://github.com/elixir-cldr/cldr_dates_times/releases/tag/v2.2.0) which fixes the dialyzer warnings.  Note that this version explicitly requires Elixir 1.8 or greater since it relies on `Calendar` functions not available in earlier releases.  This is not a change to the code, just a change to the requirement.

Please note that you will also need to update to `ex_cldr` [version 2.8.1](https://hex.pm/packages/ex_cldr/2.8.1) since I needed to change the `@spec` for `Cldr.Substitution.substitute/2` as well.

I have also checked each of the `cldr_*` family and confirmed that there are no `dialyzer` warnings issued.   Please do let me know if you see otherwise.

Thanks for keeping me honest on this, and for your feedback and support.
